app/uk/gov/hmrc/cgtpropertydisposalsfrontend/services/onboarding/IvService.scala

```diff
@@ -18,64 +18,43 @@ package uk.gov.hmrc.cgtpropertydisposalsfrontend.services.onboarding
 
 import cats.data.EitherT
 import cats.instances.future._
-import cats.instances.int._
-import cats.syntax.either._
-import cats.syntax.eq._
 import com.google.inject.{ImplementedBy, Inject, Singleton}
 import play.api.libs.json.{Json, Reads}
-import play.mvc.Http.Status.OK
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.connectors.onboarding.IvConnector
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.metrics.Metrics
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.Error
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.models.onboarding.iv.IvErrorStatus
-import uk.gov.hmrc.cgtpropertydisposalsfrontend.services.onboarding.IvServiceImpl.IvStatusResponse
-import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.HttpResponseOps._
 import uk.gov.hmrc.http.HeaderCarrier
 
 import java.util.UUID
 import scala.concurrent.{ExecutionContext, Future}
+import scala.util.chaining.scalaUtilChainingOps
 
 @ImplementedBy(classOf[IvServiceImpl])
 trait IvService {
-
-  def getFailedJourneyStatus(journeyId: UUID)(implicit
-    hc: HeaderCarrier
-  ): EitherT[Future, Error, IvErrorStatus]
-
+  def getFailedJourneyStatus(journeyId: UUID)(implicit hc: HeaderCarrier): EitherT[Future, Error, IvErrorStatus]
 }
 
 @Singleton
-class IvServiceImpl @Inject() (connector: IvConnector, metrics: Metrics)(implicit
-  ec: ExecutionContext
-) extends IvService {
+class IvServiceImpl @Inject() (connector: IvConnector, metrics: Metrics)(implicit ec: ExecutionContext)
+    extends IvService {
 
   override def getFailedJourneyStatus(
     journeyId: UUID
   )(implicit hc: HeaderCarrier): EitherT[Future, Error, IvErrorStatus] = {
     val timer = metrics.ivGetFailedJourneyStatusTimer.time()
     connector
       .getFailedJourneyStatus(journeyId)
-      .subflatMap[Error, IvErrorStatus] { response =>
-        timer.close()
-        if (response.status === OK) {
-          response
-            .parseJSON[IvStatusResponse]()
-            .bimap(
-              { e =>
-                metrics.ivGetFailedJourneyStatusErrorCounter.inc()
-                Error(e)
-              },
-              r => IvErrorStatus.fromString(r.result)
-            )
-        } else {
+      .bimap(
+        _.tap { _ =>
+          timer.stop()
           metrics.ivGetFailedJourneyStatusErrorCounter.inc()
-          Left(
-            Error(
-              s"Call to check status of failed IV journey came back with status ${response.status}"
-            )
-          )
+        },
+        r => {
+          timer.stop()
```

Comment:
looking in to the last commit, timer.stop() wasn't used here before, why are we adding it now.

Reply:
Timer.close() is a `Disposable` wrapper on top of `timer.stop()`. close calls stop, which was called explicitly in the original.